packages/client/modules/userDashboard/components/OrgUserRow/OrgMemberRow.tsx

```diff
@@ -90,29 +96,157 @@ const BillingLeaderActionMenu = lazyPreload(
+    import(/* webpackChunkName: 'OrgAdminActionMenu' */ '../../../../components/OrgAdminActionMenu')
+)
 
 const RemoveFromOrgModal = lazyPreload(
   () =>
     import(/* webpackChunkName: 'RemoveFromOrgModal' */ '../RemoveFromOrgModal/RemoveFromOrgModal')
 )
 
+interface UserAvatarProps {
+  picture?: string
+}
+
+const UserAvatar: React.FC<UserAvatarProps> = ({picture}) => (
+  <AvatarBlock>
+    {picture ? (
+      <Avatar hasBadge={false} picture={picture} size={44} />
+    ) : (
+      <img alt='' src={defaultUserAvatar} />
+    )}
+  </AvatarBlock>
+)
+
+interface UserInfoProps {
+  preferredName: string
+  email: string
+  isBillingLeader: boolean
+  isOrgAdmin: boolean
+  inactive: boolean | null
+  newUserUntil: string
+}
+
+const UserInfo: React.FC<UserInfoProps> = ({
+  preferredName,
+  email,
+  isBillingLeader,
+  isOrgAdmin,
+  inactive,
+  newUserUntil
+}) => (
+  <StyledRowInfo>
+    <RowInfoHeader>
+      <RowInfoHeading>{preferredName}</RowInfoHeading>
+      {isBillingLeader && <RoleTag>Billing Leader</RoleTag>}
+      {isOrgAdmin && <BaseTag className='bg-gold-500 text-white'>Org Admin</BaseTag>}
+      {inactive && !isBillingLeader && !isOrgAdmin && <InactiveTag>Inactive</InactiveTag>}
+      {new Date(newUserUntil) > new Date() && <EmphasisTag>New</EmphasisTag>}
+    </RowInfoHeader>
+    <RowInfoLink href={`mailto:${email}`} title='Send an email'>
+      {email}
+    </RowInfoLink>
+  </StyledRowInfo>
+)
+
+interface UserActionsProps {
+  isViewerOrgAdmin: boolean
+  isViewerBillingLeader: boolean
+  isViewerLastOrgAdmin: boolean
+  isViewerLastBillingLeader: boolean
+  organization: OrgMemberRow_organization$data
+  organizationUser: OrgMemberRow_organizationUser$data
+  preferredName: string
+  viewerId: string
+}
+
+const UserActions: React.FC<UserActionsProps> = ({
+  isViewerOrgAdmin,
+  isViewerBillingLeader,
+  isViewerLastOrgAdmin,
+  isViewerLastBillingLeader,
+  organizationUser,
+  organization,
+  preferredName,
+  viewerId
+}) => {
+  const {orgId} = organization
+  const {
+    user: {userId}
+  } = organizationUser
+  const {togglePortal, originRef, menuPortal, menuProps} = useMenu(MenuPosition.UPPER_RIGHT)
+  const {togglePortal: toggleLeave, modalPortal: leaveModal} = useModal()
+  const {togglePortal: toggleRemove, modalPortal: removeModal} = useModal()
+  const actionMenuProps = {
+    menuProps,
+    originRef,
+    togglePortal,
+    toggleLeave,
+    toggleRemove,
+    isViewerLastOrgAdmin,
+    isViewerLastBillingLeader,
+    organization,
+    organizationUser
+  }
+
+  const showLeaveButton = !isViewerOrgAdmin && !isViewerBillingLeader && viewerId === userId
+
+  return (
+    <RowActions>
+      <ActionsBlock>
+        {showLeaveButton && (
+          <StyledFlatButton onClick={toggleLeave} onMouseEnter={LeaveOrgModal.preload}>
+            Leave Organization
+          </StyledFlatButton>
+        )}
+        {(isViewerOrgAdmin || (isViewerBillingLeader && !isViewerLastBillingLeader)) && (
```

Comment:
This (and below) was the most significant change I made to what was started:

I created a wholly new control for Org Admin actions vs. actions that can be taken by the BillingLeader. This helped simplify logic when decided what role can take what action